websites/T/Typer.io/presence.ts

```diff
@@ -0,0 +1,243 @@
+const presence = new Presence({
+    clientId: "854448403273351202"
+  }),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: browsingStamp
+  };
+
+  // typer.io/play presence (aka quick play)
+  if (document.location.pathname.startsWith("/play")) {
+    const playingStatus = document
+      .querySelector(
+        "#__next > div.Play_root__16QtH > div.Play_container__392yl.false > div.Status_root__2iFRH > div > h3"
+      )
+      .textContent.split("Jump to")[0]
+      .trim();
+
+    if (playingStatus) {
+      presenceData.details = "In a quick play race:";
+      presenceData.state = playingStatus;
+    }
+
+    try {
+      const quickplayWPMUnmerged = document
+          .querySelector("#PreMiD-WPM")
+          .textContent.split("Jump to")[0]
+          .trim(),
+        quickplayWPM = `${quickplayWPMUnmerged} WPM`,
+        quickplayAccuracyUnmerged = document
+          .querySelector("#PreMiD-ACC")
+          .textContent.split("Jump to")[0]
+          .trim(),
+        quickplayAccuracy = `${quickplayAccuracyUnmerged} acc`,
+        quickplayRaceStatus = document
+          .querySelector(
+            "#__next > div.Play_root__16QtH > div.Play_container__392yl.false > div.Status_root__2iFRH > div > h3"
+          )
+          .textContent.split("Jump to")[0]
+          .trim();
+
+      if (quickplayRaceStatus === "GO!") {
+        presenceData.details = "In a quick play race:";
+        presenceData.state = `${quickplayWPM}, ${quickplayAccuracy}`;
+      }
+
+      const quickplayRacePlace = document
+          .querySelector("#PreMiD-RANK")
+          .textContent.split("Jump to")[0]
+          .trim(),
+        quickplayRaceTime = document
+          .querySelector("#PreMiD-TIME")
+          .textContent.split("Jump to")[0]
+          .trim();
+
+      if (quickplayRaceTime !== "--:--") {
+        presenceData.details = `Placed ${quickplayRacePlace} in a quick play race!`;
+        presenceData.state = `${quickplayWPM}, ${quickplayAccuracy}, ${quickplayRaceTime}`;
+      }
+
+      if (
+        quickplayRacePlace === "-" &&
+        quickplayWPM !== "undefined" &&
+        quickplayRaceStatus === "Game has Ended"
+      ) {
+        presenceData.details = "The game ended, did not place.";
```

Comment:
Did not place? This sounds wrong to me

Reply:
I can change it to "did not finish" if that makes more sense.

Reply:
yeah well, the context is that they didn't finish the game, is that right

Reply:
Yep. Fixing.